pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -2243,6 +2243,176 @@ def is_hermitian(self):
         return True
 
 
+class IntegerComparator(Operation):
```

Comment:
I wonder if this would make sense as an "arithmetic op" instead like `QubitCarry` and `QubitSum`?